src/main/java/com/spotify/folsom/ConnectFuture.java

```diff
@@ -35,11 +34,11 @@ private ConnectFuture(ObservableClient client, boolean awaitedState) {
     check(client);
   }
 
-  public static ListenableFuture<Void> disconnectFuture(ObservableClient client) {
+  public static CompletableFuture<Void> disconnectFuture(ObservableClient client) {
```

Comment:
Hm, here it might make sense to keep CompletableFuture instead of CompletionStage though.